packages/ckeditor5-cloud-services/src/token/token.ts

```diff
@@ -112,14 +114,29 @@ export default class Token extends /* #__PURE__ */ ObservableMixin() {
 				}
 
 				return this as InitializedToken;
+			} )
+			.catch( err => {
+				/**
+				 * TODO
+				 *
+				 * @error token-refresh-failed
+				 */
+				console.warn( 'token-refresh-failed: TODO' );
+
+				// If the refresh failed, keep trying to refresh the token. Failing to do so will eventually
+				// lead to the disconnection from the RTC service and the editing session (and potential data loss
+				// if the user keeps editing).
+				this._registerFailedRefreshTokenTimeout();
```

Comment:
I wonder if this should engage only when using the default refresh callback. A custom integrator's callback could come with its own refresh logic and this would be redundant or even buggy.

Reply:
I figured out that it's unlikely for a custom refresh callback to come with a retry logic and integrators who fall into this category should also benefit from auto-retry logic.